scanner.go

```
@@ -279,6 +279,16 @@ func (s *Scanner) scanNumber() (Pos, Token, string) {
 		}
 	}
 
+	if s.peek() == 'x' || s.peek() == 'X' {
```

This is not quite the way I would code it. You seem to be relying on the code above to pick up the leading '0', but the logic doesn't enforce that.  4x292 would be interpreted as hex too, right? 

I would code it something like this:

- if it's a digit, next check if it's a zero.
- if it's a zero, then start hex processing. If that fails, it's an error (right?)
- if it's not a zero, then do what was done before.

---

Yeah, you are right. I have modified it but due to lack of error handling in scanner i cant enforce the 16 significant digits or `0x` being invalid. Moreover, `4xe3` still gets scanned but results in `4`. I am not a fan of silent failures so we would need to refactor the scanner to add error handling.